File: Week 6/C Sharp/Aliens/Aliens/Models/Alien.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace Aliens.Models
{
    internal class Alien
    {
        // Properties
        public bool Dead { get; set; }
        public bool Dodging { get; set; }
        public int Health { get; set; }

        // Class constructor
        public Alien(bool dodging, int alienHealth)
        {
            //TODO: Construct the Alien object and initialise the data members appropriately.

        }

        // Methods
        //TODO: Implement appropriate getters and setters for the class

        public void Dodge()
        {
            //TODO: Implement the method
            /*Set dodging to true if the alien is alive*/

        }

        public void Hit()
        {
            //TODO: Implement the method
            /*Reduces the alien health by 1. Minimum Alien health should 0.*/
        }

        public void Miss()
        {
            //TODO: Implement the method
            /*Sets dodging to false*/

        }

    }
}

File: Week 6/Java/src/com/company/Alien.java
/*
 * Classname: Alien.java
 *
 * Version information: 1.0
 *
 * Date: 21/01/2021
 *
 * Description: The class implements an Alien for a game. The health of the alien object determines if the alien is
 * alive and or dodging. If the health attribute is 0 then the alien attribute 'dead' MUST be true. The alien cannot be
 * dodging if they are dead. Likewise, the alien cannot be dead if they health value is >0.
 * Author: Dr. Shane Wilson
 *
 */

package com.company;

public class Alien {
    //Data members
    private boolean dead;
    private boolean dodging;
    private int alienHealth;

    //Class constructors
    public Alien(boolean dodging, int alienHealth) {
        //TODO: Construct the Alien object and initialise the data members appropriately.

    }

    // Methods
    //TODO: Implement appropriate getters and setters for the class

    public void dodge() {
        //TODO: Implement the method
        /*Set dodging to true if the alien is alive*/

    }

    public void hit() {
        //TODO: Implement the method
        /*Reduces the alien health by 1. Minimum Alien health should 0.*/
    }

    public void miss() {
        //TODO: Implement the method
        /*Sets dodging to false*/

    }

}
